src/Utilities/ErrorHandling/Error.hpp

```diff
@@ -66,7 +68,29 @@ template <typename F>
     if (__builtin_is_constant_evaluated()) {                                 \
       throw std::runtime_error("Failed");                                    \
     } else {                                                                 \
-      Error_detail::abort_without_fpes(                                      \
+      Error_detail::abort_without_fpes<SpectreError>(                        \
+          __FILE__, __LINE__, static_cast<const char*>(__PRETTY_FUNCTION__), \
+          [&]() -> std::string {                                             \
+            return MakeString{} << std::setprecision(18) << std::scientific  \
+                                << m;                                        \
+          });                                                                \
+    }                                                                        \
+  } while (false)
+
+/*!
+ * \ingroup ErrorHandlingGroup
+ * \brief Same as ERROR but will throw `EXCEPTION_TYPE` instead of
+ * `SpectreError`.
+ *
+ * \note Any exception types used must have an associated explicit
+ * instantiation in `Utilities/ErrorHandling/AbortWithErrorMessage.cpp`
+ */
+#define ERROR_WITH(m, EXCEPTION_TYPE)                                        \
```

Comment:
Quick note: Catch2 calls the corresponding check `CHECK_THROWS_AS( expression, exception type )`, so you could do `ERROR_AS` to be consistent with that (if you want)

Reply:
Seems reasonable!